File: react-client/src/features/recipes/components/recipe-list/recipe-element/RecipeElement.js
import React, { Component } from "react";
import Style from "./RecipeElement.module.scss";

export default class RecipeElement extends Component {
  updateSelectedRecipe = () => {
    this.props.updateSelectedRecipe(this.props.recipe.id);
  };

  render() {
    return (
      <div
        onClick={this.updateSelectedRecipe}
        className={"d-flex flex-column bg-light " + Style.container}
      >
        <div>
          <img
            alt="recipe"
            width="320"
            src={require(`../../../../../assets/img/recipe${
              this.props.index + 1
            }.jpg`)}
          />
        </div>
        <div className="p-3 text-center">
          <h5>{this.props.recipe.name}</h5>
          <hr className="w-100" />
          <span className="text-secondary">
            {this.props.recipe.tags.join(" | ")}
          </span>
        </div>
      </div>
    );
  }
}

File: client/src/main/scala/com/lunatech/goldenalgo/onboarding/features/recipes/components/RecipeElement.scala
package com.lunatech.goldenalgo.onboarding.features.recipes.components

import com.lunatech.goldenalgo.onboarding.model.Recipe
import com.lunatech.goldenalgo.onboarding.router.AppRouter
import japgolly.scalajs.react.ScalaComponent
import japgolly.scalajs.react.component.Scala.Unmounted
import japgolly.scalajs.react.extra.router.RouterCtl
import japgolly.scalajs.react.vdom.html_<^._

object RecipeElement {

  type ImgNumber = Int

  case class Props(ctl: RouterCtl[AppRouter.Page], recipe: Recipe, imgNumber: ImgNumber)

  private val component = ScalaComponent.builder[Props]("RecipeElement")
    .render_P(props =>
      <.div(
        ^.className := "recipe-element",
        <.div(
          ^.className := "container d-flex flex-column align-items-center bg-light",
          <.img(
            ^.alt := "recipe",
            ^.width :="320",
            ^.src := s"client/src/main/resources/static/assets/img/recipe${props.imgNumber}.jpg"
          ),
          <.div(
            ^.className := "p-3 text-center",
            <.h5(props.recipe.name),
            <.hr(
              ^.className := "w-100"
            ),
            <.span(
              ^.className := "text-secondary ",
              props.recipe.tags.mkString(" | ")
            )
          ),
        ),
        props.ctl setOnClick AppRouter.RecipeDetailsPage(props.recipe.id, props.imgNumber),
      )
    ).build

  def apply(props: Props): Unmounted[Props, Unit, Unit] = component(props)

}
